File: testers/avro/java-json/generated-and-checked-in/com/example/events/OrderId.java
package com.example.events;

import com.fasterxml.jackson.annotation.JsonValue;

/** Unique order identifier */
public record OrderId(@JsonValue String value) {
  public OrderId withValue(String value) {
    return new OrderId(value);
  }

  @Override
  public java.lang.String toString() {
    return value.toString();
  }

  /** Create a OrderId from a raw value */
  public static OrderId valueOf(String v) {
    return new OrderId(v);
  }

  /** Get the underlying value */
  public String unwrap() {
    return this.value();
  }
}

File: testers/avro/scala-json/generated-and-checked-in/com/example/events/OrderId.scala
package com.example.events

import com.fasterxml.jackson.annotation.JsonValue

/** Unique order identifier */
case class OrderId(@JsonValue value: String) extends scala.AnyVal {
  /** Get the underlying value */
  def unwrap: String = {
    return this.value
  }
}

object OrderId {
  /** Create a OrderId from a raw value */
  def valueOf(v: String): OrderId = {
    return new OrderId(v)
  }
}